docs/spfx/toolchain/implement-ci-cd-with-azure-devops.md

```diff
@@ -0,0 +1,207 @@
+
+Setting up Azure DevOps for Continuous Deployments with a SharePoint Framework solution requires the following steps:
+1. Creating the Release Definition
+2. Linking the Build Artifact
+3. Creating the Environment
+4. Installing NodeJS
+5. Installing the Office 365 CLI
+6. Connecting to the App Catalog
+7. Adding the Solution Package to the App Catalog
+8. Deploying the Application
+9. Setting the Variables for the Environment
+
+### Creating the Release Definition
+Start by creating a new Release Definition with an empty template. A Release Defition is a process that is used to identify the following elements for the deployment:
+- Environment
+- Deployment tasks
+- Build artifacts
+![creating the release definition](../../images/azure-devops-spfx-10.png)
+
+### Linking the Build Artifact
+Click on `Add an artifact` and select the build definition you previously created.  Write down the `Source Alias` name you set, as you will need to use it in subsequent tasks.
+![linking the artifacts](../../images/azure-devops-spfx-11.png)
+
+### Creating the Environment
+When you create your continuous deployment environment, you can give a name and configure pre-deployment approvals, artificats filters (i.e. deploy only if the build comes from this or that branch), and much more by clicking on the buttons around the environment box or directly on the title.
+![creating the environment](../../images/azure-devops-spfx-12.png)
+
+### Installing NodeJS
+By click on `1 job, 0 tasks` you can access the tasks configuration view, which works similarly to the build definition. Here, you can select the set of tasks that will run only for this specific environment.  This includes installing NodeJS version 8 or later.
+Add a `Node tool installer` task and define `8.X` in the `Version Spec` field. 
+![installing node 8](../../images/azure-devops-spfx-13.png)
+
+### Installing the Office 365 CLI
+The Office 365 Common Language Interface (CLI) is an open source project built by the OfficeDev PnP Community. In order to leverage the  CLI as part of your Release Definition, you first need to install it.  Then, you will be able to take advantage of commands available  to handle deployment. Add a `npm` task, select a `Custom` command and type `install -g @pnp/office365-cli` in the `Command and Arguments` field.
+![installing office 365 cli](../../images/azure-devops-spfx-14.png)
+> [!NOTE] 
+> Learn more about the [Office 365 CLI](https://pnp.github.io/office365-cli/)
+
+### Connecting to the App Catalog
+Before using the App Catalog in you deployment environment, you first need to authenticate against the App Catalog of your tenant.  To do so, add a `Command Line` task and paste in the following command into the `script` field `o365 spo connect https://$(tenant).sharepoint.com/$(catalogsite) --authType password --userName $(username) --password $(password)
```

Comment:
When using the CLI you don't need to connect to a specific site. Instead you connect either to the tenant admin or the root site, depending on the commands that you want to use.

Reply:
What if they are using site collections app catalogs? http://www.andrewconnell.com/blog/managing-sharepoint-online-site-collection-app-catalogs-with-the-office-365-cli (although I haven't tested that scenario)